### DISCUSSION TITLE
part-ii/2021-23/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#342] [Question comments]
Part II 2021 Automata and Formal Languages: Paper 1, Section I, F 

https://questions.tripos.org/part-ii/2021-23/

**Comment by artie2000** [comment]
<details>
<summary>Last part:</summary>

On input $n$, enumerate $\mathbb{K}$ by a diagonal process as $\{a_0,a_1,...\}$.

If $n=a_0$, return 0 and halt.

If $n=a_k$ for $k>0$, return $a_{k-1}$ and halt.

Then let $S=\mathbb{K}\cup\{0\}$.

</details>
